### PR TITLE
fix: only adjust autoscrolling in interactive mode

### LINKED ISSUE #16098: Auto scroll is turned off sometimes
### Current behavior
As you can see in the video below, auto scroll turns off at 27s mark and then stays off for the rest of the spec: 

https://user-images.githubusercontent.com/49371/115434222-74d32500-a1d6-11eb-9f63-6b0037c46cff.mp4

It's not consistent, it happened locally more often than in CI, but the recording above is from CI.

### Desired behavior

I'd like it to stay on, as there's no way to see the commands without it.

### Versions

Cypress `7.1.0`

### LINKED ISSUE #22199: Auto Scrolling no longer works after some time
### Current behavior

In run mode and with the browser hidden.

After upgrading to Cypress >= 10, in large tests, Auto Scrolling stops working at some point.

Before:

https://user-images.githubusercontent.com/11736724/172683673-a7ed4f52-701e-424e-a7ff-64c06851aec6.mp4

After:

https://user-images.githubusercontent.com/11736724/172682939-a244a67f-5d7a-4da2-84df-f946641f7438.mp4

### Desired behavior

_No response_

### Test code to reproduce

npx cypress run

### Cypress Version

10.0.3

### Other

_No response_

### PR DESCRIPTION
**Pull request by marktnoonan**
- Closes #16098 
- Closes #22199

### User facing changelog

In run mode, autoscrolling of the command log can no longer become disabled.

### Additional details

In the linked issues several users have reported issues with the command log autoscroll becoming disabled in run mode intermittently, or in some cases every time. This is particularly painful when reviewing the video for a failed test, as without autoscrolling, the error message associated with the failure is not visible. This makes debugging a lot harder.

The underlying cause is that for whatever reason, especially when resources are low, the browser sometimes seems to incorrectly detect _user_ scrolling when elements are rapidly being added to the command log and Cypress is autoscrolling, which triggers the `temporarilySetAutoScrolling(false)` code that is helpful in open mode, since in open mode it's natural for  to manually scroll the reporter while tests are running, and autoscroll would contradict that.

Since this has no purpose in run mode (or a very narrow one, when running with the `--headed` flag), it seems best to just turn off the scroll handling when we know that any scrolling is not possibly coming from the user.

### Steps to test

This one is difficult to test manually as we have almost never been able to reproduce it. But users who have tried pre-release binaries implementing the fix report that it works.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by lmiller1990** [comment]
Going to add @emilyrohrbough who has been working on the reporter a lot lately for some additional 👀

**Review by lmiller1990 — CHANGES_REQUESTED** [review]

File: packages/reporter/src/runnables/runnables.tsx [hunk 1]
```diff
@@ -176,11 +176,31 @@ class Runnables extends Component<RunnablesProps> {
...
-    }))
+    let maybeHandleScroll: UserScrollCallback | undefined = undefined
+
+    const containerEl = this.refs.container as HTMLElement
+
+    if (window.__CYPRESS_MODE__ === 'open') {
+      // in open mode, listen for scroll events so that users can pause the command log auto-scroll
+      // by manually scrolling the command log
+      containerEl.setAttribute('data-cy-scroll-listen', 'true')
+      maybeHandleScroll = action('user:scroll:detected', () => {
```
Comment (lmiller1990) [inline packages/reporter/src/runnables/runnables.tsx:187]: Took me a while to figure out these data attrs are purely for testing. I don't think this is ideal - I think they are okay to make selecting elements less brittle, like `findByTestId(...).click()`, as opposed to `findById("#blah")`.

What I think we really want to do here is spy on the function and assert it was/was not called with the correct arguments (next best thing to actually adding a regression test, but as you pointed out this is really hard to reproduce in a reliable way).

While writing a comment on how we might test this by spying on the method, I ended up just giving it at try, as well as killing some technical debt along the way, what do you think?

https://github.com/cypress-io/cypress/pull/23774

Comment (marktnoonan) [reply packages/reporter/src/runnables/runnables.tsx:184]: oh nice, this looks really good, merged!

**Comment by marktnoonan** [comment]
Updated with @lmiller1990's much better tests 🙏

**Review by mike-plummer — APPROVED** [review]
Verified scrolling is working as expected in Electron, Chrome, FF, and no issues in headless. Looks good 👍

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**